### PR TITLE
Clarify `UndoRedo` `MergeMode` and `add_*_reference` docs

### PR DESCRIPTION
**Pull request by Jordyfel**
Clarifies parts of the docs which I had trouble understanding when using the class.

### CODE REVIEW

File: doc/classes/UndoRedo.xml [hunk 1]
```diff
@@ -272,10 +276,10 @@
 			Makes "do"/"undo" operations stay in separate actions.
 		</constant>
 		<constant name="MERGE_ENDS" value="1" enum="MergeMode">
-			Makes so that the action's "undo" operations are from the first action created and the "do" operations are from the last subsequent action with the same name.
+			Merge this action with the prevoius one if they have the same name. Keep only the first action's "undo" operations and the last action's "do" operations. Useful for sequential changes to a single value.
```
Comment (AThousandShips) [inline doc/classes/UndoRedo.xml:279]:
```suggestion
			Merge this action with the previous one if they have the same name. Keep only the first action's "undo" operations and the last action's "do" operations. Useful for sequential changes to a single value.
```

File: doc/classes/UndoRedo.xml [hunk 2]
```diff
@@ -272,10 +276,10 @@
...
 		</constant>
 		<constant name="MERGE_ALL" value="2" enum="MergeMode">
-			Makes subsequent actions with the same name be merged into one.
+			Merge this action with the prevoius one if they have the same name.
```
Comment (AThousandShips) [inline doc/classes/UndoRedo.xml:282]:
```suggestion
			Merge this action with the previous one if they have the same name.
```

File: doc/classes/UndoRedo.xml [hunk 3]
```diff
@@ -112,11 +112,13 @@
...
 			<param index="0" name="object" type="Object" />
 			<description>
-				Register a reference for "do" that will be erased if the "do" history is lost. This is useful mostly for new nodes created for the "do" call. Do not use for resources.
+				Register a reference to an object that will be erased if the "do" history is deleted. This is useful for objects added by the "do" action and removed by the "undo" action.
+				When the "do" history is deleted, if the object is a [RefCounted], it will be unreferenced. Otherwise, it will be freed. Do not use for resources.
 				[codeblock]
 				var node = Node2D.new()
 				undo_redo.create_action("Add node")
 				undo_redo.add_do_method(add_child.bind(node))
+				# Free the node when the "do" history is deleted.
```
Comment (KoBeWi) [inline doc/classes/UndoRedo.xml:121]: This comment just repeats what is already said above.

File: doc/classes/UndoRedo.xml [hunk 4]
```diff
@@ -272,10 +276,10 @@
 			Makes "do"/"undo" operations stay in separate actions.
 		</constant>
 		<constant name="MERGE_ENDS" value="1" enum="MergeMode">
-			Makes so that the action's "undo" operations are from the first action created and the "do" operations are from the last subsequent action with the same name.
+			Merge this action with the previous one if they have the same name. Keep only the first action's "undo" operations and the last action's "do" operations. Useful for sequential changes to a single value.
```
[the remark this answers is not in the document]
Comment (KoBeWi) [reply doc/classes/UndoRedo.xml:279]: I'd say "merge" is correct in this case, depending if you want to describe what will happen vs what this value does.

**Review by KoBeWi — APPROVED** [review]
I think the added code comments are redundant, but the rewording is correct.

File: doc/classes/UndoRedo.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (AThousandShips) [reply doc/classes/UndoRedo.xml:279]: Then the one above should be changed at least, but at least "keeps" should stay

**Comment by Jordyfel** [comment]
Not sure about makes/merges, made it as it was before. Removed the comments

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**